### ISSUE TITLE
safe-get / safe-set: fix / document more differences with lodash

### ISSUE THREAD

**Issue by jakub-g** [angus-c/just#146]
1. With lodash you can pass a default value as a third parameter, with just you need to check for `undefined` manually (or ` || defaultValue` if accepable)

2. https://github.com/angus-c/just/blob/master/TRADEOFFS.md says

> i. just-get and just-set follows dotty for (obj, ['a.b']) style arguments. Lodash uses its own rules.

However the top-level and package-level docs show the examples with the array notation which seem like it would work just like lodash:

```
const obj = { a: { aa: { aaa: 2 } }, b: 4 };

get(obj, 'a.aa.aaa'); // 2
get(obj, ['a', 'aa', 'aaa']); // 2

get(obj, 'b.bb.bbb'); // undefined
get(obj, ['b', 'bb', 'bbb']); // undefined
```

The thing is however that `just` mutates the passed array via `.shift()`, so you must pass literals and avoid passing references. Example where this breaks:

```
const PATHS = [['a', 'aa'], ['b', 'bb'], ['c', 'cc']]

function doStuff(obj) { 
  PATHS.forEach((subpath) => {
    const value = get(obj, subpath)
    ...
  })
}

doStuff(obj1)
doStuff(obj2) // oops! PATHS will be array of empty arrays
```

**Comment by angus-c** [comment]
Thanks very much for this issue and PR. I think I'd like to fix the array mutation before updating TRADEOFFS—will try to get to this tomorrow.

**Comment by angus-c** [comment]
Ok I think the array mutation issue is now fixed (`just-safe-get@1.3.2`)
Up to you whether you want to update your PR accordingly, or I can do it when I get a chance.

Thanks again for this.

**Comment by wrporter** [comment]
Funny that I was just trying to use the defaultValue yesterday of safe-get. I was so confused that it was not working for me. And then I looked at the history and realized this is being added right when I needed it. Thanks for getting this in!

**Comment by angus-c** [comment]
Fixed by #151
